### PR TITLE
[IMP] hr_*, sms: improve ux

### PR DESCRIPTION
**Pull request by phwa-odoo**
Reorder smart buttons on the hr_employee to have them in the following order:
1. Documents
2. Holiday status
3. Time Off
4. Planning
5. Timesheets
6. Equipments
7. Cars
8. Contracts
9. Appraisal
10. Work Entries
11. Payroll
12. Attendance
13. Extra Hours

- Log in the chatter the name of the plan that has been started via the button "Launch Plan"

- When on mobile, open the Time Off Calendar in month view

- When sending a sms without any message, only display Message once in the invalid fields

- Avoid the sms-help info disappearing when clicking on it and there are characters in the body of the message

- Update the tooltip of phone number in the send sms wizard

task-2735679

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr/static/src/scss/hr.scss [hunk 1]
```diff
@@ -58,3 +58,9 @@
 .o_form_label.o_hr_form_label {
     cursor: default;
 }
+
+.hr_presence {
+    font-size: 0.75em;
+    vertical-align: middle;
+  }
+
```
Comment (kba-odoo) [inline addons/hr/static/src/scss/hr.scss:66]: Wrong indentation and one extra too many :innocent: 
```suggestion
}
```

**Review by kbapt — CHANGES_REQUESTED** [review]

File: addons/sms/wizard/sms_composer.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def default_get(self, fields):
     template_id = fields.Many2one('sms.template', string='Use Template', domain="[('model', '=', res_model)]")
     body = fields.Text(
         'Message', compute='_compute_body',
-        precompute=True, readonly=False, store=True, required=True)
+        precompute=True, readonly=False, store=True)
```
Comment (tde-banana-odoo) [inline addons/sms/wizard/sms_composer.py:74]: Why this change ? I don't think we want to send void SMS ...

Comment (phwa-odoo) [reply addons/sms/wizard/sms_composer.py:74]: The required is now conditional in the view: 
https://github.com/odoo/odoo/pull/83745/files#diff-dd4eeb56d7e858242e106057240dfb4ac532476372c8188f793a90ad3364bfd6R50-R51
The issue was that the field being required in the `wizard/sms_composer.py` file and the body field appearing twice in the view: 
https://github.com/odoo/odoo/pull/83745/files#diff-dd4eeb56d7e858242e106057240dfb4ac532476372c8188f793a90ad3364bfd6L51-L52
make two invalid fields error for the same field if left empty and trying to send a SMS https://imgur.com/AAX0wIq

Comment (tde-banana-odoo) [reply addons/sms/wizard/sms_composer.py:74]: Well we cannot remove the required at model level ... sending a void email is ok as you may have some other information, not sending a void sms ^^ . It that is the only condition I would say tant pis and keep the required on the field itself.

Comment (phwa-odoo) [reply addons/sms/wizard/sms_composer.py]: Got it, thanks! The field is required=True again

File: addons/hr/views/hr_employee_views.xml [hunk 3]
```diff
@@ -135,55 +118,72 @@
...
+                                                <field name="visa_no"/>
+                                                <field name="permit_no"/>
+                                            </group>
+                                            <group col="4" colspan="2">
+                                                <field name="visa_expire"/>
+                                                <field name="work_permit_expiration_date"/>
+                                            </group>
+                                            <group>
+                                                <field name="has_work_permit" widget="work_permit_upload"/>
+                                            </group>
```
Comment (kba-odoo) [inline addons/hr/views/hr_employee_views.xml:170]: I'd make this field take the whole half-column, it looks kinda weird

image

File: addons/sms/wizard/sms_composer_views.xml [hunk 4]
```diff
@@ -47,9 +47,8 @@
...
                             <field name="recipient_single_number_itf" class="oe_inline" nolabel="1" options="{'onchange_on_keydown': True}" placeholder="e.g. +1 415 555 0100"/>
                         </div>
-                                                
-                        <field name="body" widget="sms_widget" attrs="{'invisible': ['|', ('comment_single_recipient', '=', False), ('recipient_single_valid', '=', True)]}"/>
-                        <field name="body" widget="sms_widget" attrs="{'invisible': [('comment_single_recipient', '=', True), ('recipient_single_valid', '=', False)]}" default_focus="1"/>
+                        <field name="body" widget="sms_widget" attrs="{'invisible': ['|', ('comment_single_recipient', '=', False), ('recipient_single_valid', '=', True)], 'required': ['|', ('comment_single_recipient', '=', False), ('recipient_single_valid', '=', True)]}"/>
+                        <field name="body" widget="sms_widget" attrs="{'invisible': [('comment_single_recipient', '=', True), ('recipient_single_valid', '=', False)], 'required': [('comment_single_recipient', '=', True), ('recipient_single_valid', '=', False)]}" default_focus="1"/>
```
Comment (kba-odoo) [inline addons/sms/wizard/sms_composer_views.xml:51]: Those changes can be undone as the field is required again

**Review by kbapt — CHANGES_REQUESTED** [review]

File: addons/sms/wizard/sms_composer_views.xml (hunk 4 above)
Comment (phwa-odoo) [reply addons/sms/wizard/sms_composer_views.xml:51]: I was about to remove it during the last push but saw that it avoids displaying the same field twice in the invalid fields error so the required=True in the model had no effect on the desired behavior specified in the task :sweat_smile: 

image

File: addons/hr/views/hr_employee_views.xml (hunk 3 above)
Comment (phwa-odoo) [reply addons/hr/views/hr_employee_views.xml]: Good catch, thanks!

File: addons/sms/wizard/sms_composer_views.xml (hunk 4 above)
Comment (kba-odoo) [reply addons/sms/wizard/sms_composer_views.xml:51]: Ok ok makes sense now

**Comment by kba-odoo** [comment]
robodoo r+
